### PR TITLE
Print

### PR DESCRIPTION
**Pull request by jaxxzer**

### CODE REVIEW

File: examples/ping1d-advanced/ping1d-advanced.ino [hunk 1]
```diff
@@ -12,10 +12,9 @@
 #include "SoftwareSerial.h"
 
 SoftwareSerial pingSerial = SoftwareSerial(9, 10);
-HardwareSerial& debugSerial = Serial;
 
 static char _debug_buffer[200];
-#define printf(fmt, args ...)  do {sprintf(_debug_buffer, fmt "\n\r", ## args); debugSerial.print(_debug_buffer);} while(0)
```
Comment (patrickelectric) [inline examples/ping1d-advanced/ping1d-advanced.ino]: Just to make sure, should we use Serial*1* or we should create something like:
```cpp
auto debugSerial = Serial;
```

File: examples/ping1d-simple/ping1d-simple.ino [hunk 2]
```diff
@@ -34,11 +33,11 @@ void toggleLed()
 void setup()
 {
     pingSerial.begin(19200);
-    debugSerial.begin(115200);
+    Serial.begin(115200);
     pinMode(ledPin, OUTPUT);
-    debugSerial.println("Blue Robotics ping1d-simple.ino");
+    Serial.println("Blue Robotics ping1d-simple.ino");
```
[the remark this answers is not in the document]
Comment (jaxxzer) [reply examples/ping1d-simple/ping1d-simple.ino]:
```
/home/jack/Arduino/libraries/ping-arduino/examples/ping1d-simple/ping1d-simple.ino: In function 'void setup()':
ping1d-simple:35:37: error: expected ')' before '__FILENAME__'
     Serial.println("Blue Robotics " __FILENAME__ ".ino");
```

[the remark this answers is not in the document]
Comment (jaxxzer) [reply examples/ping1d-simple/ping1d-simple.ino]: if it did work, I would make the [same argument](https://github.com/bluerobotics/ping-arduino/pull/11#discussion_r249986951) to not do it in arduino examples.

File: examples/pingmessage/pingmessage.ino [hunk 3]
```diff
@@ -148,8 +152,35 @@ void loop()
...
+        Serial.println(parser.errors);
     }
+
+    // Toggle the LED to show that the program is running
+    digitalWrite(ledPin, !digitalRead(ledPin));
+}
+
+bool waitMessage(enum Ping1DNamespace::msg_ping1D_id id, uint16_t timeout_ms = 400)
+{
+    uint32_t tstart = millis();
```
Comment (patrickelectric) [inline examples/pingmessage/pingmessage.ino]: you can create a single timeout variable here.
```cpp
 uint32_t timeout_time = millis() + timeout_ms;
 while (millis() < timeout_time) {
```

Comment (jaxxzer) [reply examples/pingmessage/pingmessage.ino]: this amounts to exactly the same thing. I don't think it's easier to read.

Comment (jaxxzer) [reply examples/pingmessage/pingmessage.ino]: done

Comment (patrickelectric) [reply examples/pingmessage/pingmessage.ino]: For each while loop it needs to run one more operation to do the + operation

Comment (jaxxzer) [reply examples/pingmessage/pingmessage.ino]: I wonder if the compiler is smarter than me.

**Pull request merged**